File: Data Structures & Algorithms/two-integer-sum/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        map = {}
        for i, num in enumerate(nums):
            remainder = target - num
            if remainder in map:
                return [map[remainder], i]
            map[num] = i
        return []

File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.cs
public class Solution {
    public int[] TwoSum(int[] numbers, int target) {
        int[] result = new int[2];

        int start = 0, end = numbers.Length -1;

        while(start < end) 
        {
            int sum = numbers[start] + numbers[end];
            if(sum == target) {
                result[0] = start + 1;
                result[1] = end + 1;
                return result;
            }
            else if(sum < target) 
            {
                start ++;
            }
            else 
            {
                end --;
            }
        }

        return result;
    }
}
